From be9ae98b3da5b8bac98e342b4aee3a141f689f7c Mon Sep 17 00:00:00 2001
From: Alexander Graf <agraf@suse.de>
Date: Sun, 20 Jan 2013 23:00:39 +0100
Subject: [PATCH] ARM: fix imx usb driver build issue
Patch-Mainline: No, too hacky :)

The FSL i.MX USB drivers have issues with the new multiarch patch series.
Fix them up to access their i.MX headers directly.

This patch is only meant to get things building for now. Please do not apply
or send upstream!

Signed-off-by: Alexander Graf <agraf@suse.de>

diff --git a/drivers/usb/gadget/fsl_mxc_udc.c b/drivers/usb/gadget/fsl_mxc_udc.c
index 1b0f086..1c06d1f 100644
--- a/drivers/usb/gadget/fsl_mxc_udc.c
+++ b/drivers/usb/gadget/fsl_mxc_udc.c
@@ -18,7 +18,7 @@
 #include <linux/platform_device.h>
 #include <linux/io.h>
 
-#include <mach/hardware.h>
+#include "../../../arch/arm/mach-imx/hardware.h"
 
 static struct clk *mxc_ahb_clk;
 static struct clk *mxc_per_clk;
diff --git a/drivers/usb/gadget/imx_udc.c b/drivers/usb/gadget/imx_udc.c
index a0eb857..21d2e0a 100644
--- a/drivers/usb/gadget/imx_udc.c
+++ b/drivers/usb/gadget/imx_udc.c
@@ -36,7 +36,7 @@
 #include <linux/usb/gadget.h>
 
 #include <linux/platform_data/usb-imx_udc.h>
-#include <mach/hardware.h>
+#include "../../../arch/arm/mach-imx/hardware.h"
 
 #include "imx_udc.h"
 
